beets/dbcore/db.py

```diff
@@ -330,19 +340,10 @@ def items(self):
         for key in self:
             yield key, self[key]
 
-    def get(self, key, default=None):
-        """Get the value for a given key or `default` if it does not
-        exist.
-        """
-        if key in self:
-            return self[key]
-        else:
-            return default
-
```

Comment:
This is really just a matter of aesthetics, but one alternative would be to have `get` just do the necessary exception handling. That is, `__getitem__` would continue to contain all the "real" logic and include the `raise` at the end. Then, `get` would change to contain:

```py
try:
    return self[key]
except KeyError:
    return default
```

That would help keep the docstring for `get` sensible, for example.

Reply:
It might have been a bit of premature optimization, but afaik exception handling in Python isn't exactly performant (creation of exception object, traceback, caching of frames etc.) and combining the code paths was pretty trivial. I also liked that I could use this to have a short implementation in `Item.get`.

The docstring issue doesn't bother me personally because it properly describes what `get` can do. It *does* have a different function signature than `dict.get`, which we need for the `with_album` parameter anyway, but it's still compatible. I think I prefer `Optionally` over `Alternatively`, however.